### PR TITLE
redis: Configurable Username

### PR DESCRIPTION
**Pull request by oxzi**
The Redis ACL system was introduced with Redis 6.0. It introduced users with precisely granular permissions. This change allows Icinga DB to connect to a Redis with an ACL user.

### CODE REVIEW

File: redis/config.go [hunk 1]
```diff
@@ -55,5 +56,9 @@ func (r *Config) Validate() error {
 		return errors.New("Redis host missing")
 	}
 
+	if r.Username != "" && r.Password == "" {
+		return errors.New("A Redis username requires a password as well")
```
Comment (yhabteab) [inline redis/config.go:60]: GO errors always begin in lower case, so I would rephrase this to `'password' must be set, if username is provided` and get rid of the `A ...` part.

**Review by yhabteab — APPROVED** [review]

File: redis/config.go (hunk 1 above)
Comment (oxzi) [reply redis/config.go]: Good catch, thanks.

Comment (lippserd) [reply redis/config.go]: I think we should always put important information at the beginning of the sentence. The user has to perform an action, so I opt for Yonas' suggestion.

File: redis/config.go [hunk 2]
```diff
@@ -55,5 +56,9 @@ func (r *Config) Validate() error {
 		return errors.New("Redis host missing")
 	}
 
+	if r.Username != "" && r.Password == "" {
+		return errors.New("password must be set, if username is provided")
```
Comment (yhabteab) [inline redis/config.go:60]: Sorry, but after testing this out, the validation error on its own doesn't seem clear enough as to which password it refers to, so I would prefix it with a *lowered* `redis`.

Comment (oxzi) [reply redis/config.go:60]: With _Redis_ being a proper name, I would not put it to lowercase. Furthermore, it is written in an uppercase error just three lines above. I am going to prefix it now with "Redis", unless someone else wants to philosophize more about an error message.

**Review by yhabteab — APPROVED** [review]

**Review by lippserd — APPROVED** [review]
